api/renter.go

```diff
@@ -350,49 +378,118 @@ func (api *API) renterDeleteHandler(w http.ResponseWriter, req *http.Request, ps
 
 // renterDownloadHandler handles the API call to download a file.
 func (api *API) renterDownloadHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
-	destination := req.FormValue("destination")
-	// Check that the destination path is absolute.
-	if !filepath.IsAbs(destination) {
-		WriteError(w, Error{"destination must be an absolute path"}, http.StatusBadRequest)
+	params, err := api.parseDownloadParameters(w, req, ps)
+	if err != nil {
+		WriteError(w, Error{err.Error()}, http.StatusBadRequest)
 		return
 	}
 
-	err := api.renter.Download(strings.TrimPrefix(ps.ByName("siapath"), "/"), destination)
-	if err != nil {
-		WriteError(w, Error{"download failed: " + err.Error()}, http.StatusInternalServerError)
-		return
+	if params.Async { // Create goroutine if `async` param set.
+		// check for errors for 5 seconds to catch validation errors (no file with
+		// that path, invalid parameters, insufficient hosts, etc)
+		errchan := make(chan error)
+		go func() {
+			errchan <- api.renter.Download(params)
+		}()
+		select {
+		case err = <-errchan:
+			if err != nil {
+				WriteError(w, Error{"download failed: " + err.Error()}, http.StatusInternalServerError)
+				return
+			}
+		case <-time.After(time.Second * 5):
```

Comment:
Five seconds is definitely way too long for an async call. It should be waiting maybe 100ms, but even that is probably too much. 20ms if we have faith in the program being able to validate quickly.

We could always add more strict execution. Rather than wait until a certain timeout, we could block on a channel that the call sends down after validation is complete. But let's save that for another PR.